src/main/java/com/legendary/coffeeShop/dao/repositories/ProductRepository.java

```diff
@@ -10,4 +10,6 @@
     Product findByProductTypeAndStatus(String productType, ProductStatus status);
 
     Set<Product> findAllByStatus(ProductStatus status);
+
+    Product findById(int id);
```

Comment:
It's not necessary - we don't know the id of the db record it's a generated number